test/e2e/e2e_test.go

```diff
@@ -252,45 +293,260 @@ var _ = Describe("e2e external gateway connectivity", func() {
+
+		// Attempt to retrieve the pod IP that will run the external tunnel endpoint
+		kubectlOut, err := framework.RunKubectl("get", "pods", ovnNsFlag, "-l", labelFlag, jsonFlag, fieldSelectorControlNodeFlag)
+		if err != nil {
+			framework.Failf("Expected container %s running on %s error %v", ovnContainer, ovnControlNode, err)
+		}
+		extGatewayPodName = strings.Trim(kubectlOut, "'")
+
+		// retrieve the ip of the node where the vxlan interface will be used as a gateway
+		extGatewayIP := framework.RunKubectlOrDie("get", "pods", extGatewayPodName, ipFlag, ovnNsFlag)
+		// Annotate the test namespace
+		annotationFlag := fmt.Sprintf("k8s.ovn.org/hybrid-overlay-external-gw=%s", extGatewayIP)
+		framework.Logf("Annotating the external gateway test namespace")
+		framework.RunKubectlOrDie("annotate", "namespace", f.Namespace.Name, annotationFlag)
+
+		// Attempt to retrieve the pod name that will source the tunnel test in non-HA mode
+		kubectlOut, err = framework.RunKubectl("get", "pods", ovnNsFlag, "-l", labelFlag, jsonFlag, fieldSelectorFlag)
+		if err != nil {
+			framework.Failf("Expected container %s running on %s error %v", ovnContainer, ovnWorkerNode2, err)
+		}
+		haMode = false
+		// Attempt to retrieve the pod name that will source the tunnel test in HA mode
+		if kubectlOut == "''" {
+			haMode = true
+			kubectlOut, err = framework.RunKubectl("get", "pods", ovnNsFlag, "-l", labelFlag, jsonFlag, fieldSelectorHaFlag)
+			if err != nil {
+				framework.Failf("Expected container %s running on %s error %v", ovnContainer, ovnHaWorkerNode3, err)
+			}
+		}
 
-	// Cleanup the external interface after the test has completed
-	AfterEach(func() {
-		framework.Logf("Tearing down interface %s on %s", macvlanIface, ovnPodName)
-		framework.RunKubectlOrDie("exec", ovnPodName, ovnNsFlag, ovnContainerFlag,
-			"--", "ip", "link", "delete", macvlanIface)
 	})
 
-	It("Should validate connectivity to the macvlan interface which is simulating an external gateway", func() {
-		// HA CI mode runs a named set of nodes with a prefix of ovn-control-plane
+	It("Should validate connectivity to the vxlan interface which is simulating an external gateway and validate traffic was encapsulated and sent", func() {
+		// non-ha ci mode runs a set of kind nodes prefixed with ovn-worker
+		ciWorkerNodeSrc := ovnWorkerNode2
 		if haMode {
-			By(fmt.Sprintf("Creating a container on %s and verifying connectivity to the external gateway on %s", ovnHaWorkerNode2, ovnHaWorkerNode))
-			framework.ExpectNoError(
-				checkConnectivityPingToHost(f, ovnHaWorkerNode2, "ext-gateway-ci", pingTarget, IPv4PingCommand, 30))
+			// ha ci mode runs a named set of nodes with a prefix of ovn-control-plane
+			ciWorkerNodeSrc = ovnHaWorkerNode3
+		}
+		// Get the current rx counter from the vxlan interface prior to the test
+		procDir := fmt.Sprintf("/sys/class/net/vxlan_sys_%s/statistics/rx_packets", vxlanPort)
```

Comment:
where do you actually setup this vxlan interface?

Reply:
OVS sets up a netlink interface named `vxlan_sys_4789` when the VXLAN tunnel is defined so I'm just using that one in the patch rather than setting up new nodes to test to. Same results so I figured less was better here. Good catch, I should have mentioned that in the comments.

Reply:
ah ok so you are just sending to the vxlan tunnel on a control node and treating that as the other end of the vtep? Good idea. Just add those comments so its obvious to everyone. Thanks.